save-demo-cpg/src/main/kotlin/com/saveourtool/save/demo/cpg/controller/CpgController.kt

```diff
@@ -0,0 +1,191 @@
+import io.swagger.v3.oas.annotations.tags.Tag
+import io.swagger.v3.oas.annotations.tags.Tags
+import org.apache.commons.io.FileUtils
+import org.neo4j.driver.exceptions.AuthenticationException
+import org.neo4j.ogm.config.Configuration
+import org.neo4j.ogm.exception.ConnectionException
+import org.neo4j.ogm.session.Session
+import org.neo4j.ogm.session.SessionFactory
+import org.slf4j.LoggerFactory
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.io.File
+import java.nio.file.Files.createTempDirectory
+import java.nio.file.Path
+import java.util.*
+import kotlin.io.path.*
+import kotlin.system.exitProcess
+
+const val FILE_NAME_SEPARATOR = "==="
+
+
+private const val S_TO_MS_FACTOR = 1000
+private const val TIME_BETWEEN_CONNECTION_TRIES: Long = 2000
+private const val MAX_COUNT_OF_FAILS = 10
+private const val EXIT_SUCCESS = 0
+private const val EXIT_FAILURE = 1
+private const val VERIFY_CONNECTION = true
+private const val DEBUG_PARSER = true
+private const val AUTO_INDEX = "none"
+private const val PROTOCOL = "bolt://"
+
+private const val DEFAULT_HOST = "localhost"
+private const val DEFAULT_PORT = 7687
+private const val DEFAULT_USER_NAME = "neo4j"
+private const val DEFAULT_PASSWORD = "123"
+private const val DEFAULT_SAVE_DEPTH = -1
+
+/**
+ * A simple controller
+ */
+@ApiSwaggerSupport
+@Tags(
+    Tag(name = "cpg-demo"),
+)
+@RestController
+@RequestMapping("/cpg/api")
+class CpgController {
+    private val logger = LoggerFactory.getLogger(CpgController::class.java)
+
+    /**
+     * @param language
+     * @param sourceCode
+     * @return result of uploading, it contains ID to request the result further
+     */
+    @PostMapping("/upload-code")
+    @OptIn(ExperimentalPython::class)
+    fun uploadCode(
+        @RequestBody request: DemoRunRequest,
+    ) = blockingToMono {
+        val tmpFolder = createTempDirectory(request.params.language.modeName)
+        try {
+            createFiles(request, tmpFolder)
+
+            val translationConfiguration =
+                TranslationConfiguration.builder()
+                    .topLevel(null)
+                    .defaultLanguages()
+                    .registerLanguage(PythonLanguageFrontend::class.java, listOf(".py"))
+                    .debugParser(true)
+                    .sourceLocations(tmpFolder.toFile())
+                    .defaultPasses()
+                    .inferenceConfiguration(
+                        InferenceConfiguration.builder()
+                            .inferRecords(true)
+                            .build()
+                    )
+                    .build()
+
+            val result = TranslationManager.builder()
+                .config(translationConfiguration)
+                .build()
+                .analyze()
+                .get()
+
+            saveTranslationResult(result)
+        } finally {
+            FileUtils.deleteDirectory(tmpFolder.toFile())
```

Comment:
Will it recursively delete all the files in folder?

Reply:
yes, it's apache utilities